### ISSUE TITLE
`useless-try-except (TRY203)` false negative?

### ISSUE THREAD

**Issue by harupy** [astral-sh/ruff#16561]
### Summary

X-post from https://github.com/guilatrova/tryceratops/issues/76. In the following code, should the last except handler be flagged as a `useless-try-except` violation?

```python
try:  
    foo()  
except ValueError:  
    print("Do not throw")  
except TypeError:  
    raise
```

Playground reproduction: https://play.ruff.rs/44084483-f2df-424a-87ca-dc5d6884aebe

**Comment by harupy** [comment]
@InSyncWithFoo by the second handler, you mean `except ValueError`?

**Comment by harupy** [comment]
I see, so:

```
class CustomError(ValueError, TypeError): ...

try:
    raise CustomError
except ValueError:
    print('ValueError')               # This is reached
except TypeError:
    raise
```

isn't equivalent to:

```
class CustomError(ValueError, TypeError): ...

try:
    raise CustomError
except ValueError:
    print('ValueError')               # This is reached
```

**Comment by harupy** [comment]
Thanks for the confirmation :）

**Comment by harupy** [comment]
Can I file a PR to fix this?

**Comment by MichaReiser** [comment]
It seems this is intentional, see https://github.com/astral-sh/ruff/pull/4461#discussion_r1195753624

**Comment by harupy** [comment]
@MichaReiser Yes, that's what I'm suggesting.

```python
try:
    ...
except Exception:
    pass # swallow the error
except ValueError:
    raise # likely a mistake, should be moved above the first handler, or removed if unnecessary
```
